src/chol_cache.h

```diff
@@ -43,6 +49,27 @@ struct lower_chol_nonspatial: public lower_chol_base<Type> {
       return Li;
     }
   }
+  matrix<Type> get_sigma(std::vector<int> visits, matrix<Type> dist) {
+    auto target = this->sigmas.find(visits);
+    if (target != this->sigmas.end()) {
+      return target->second;
+    } else {
+      matrix<Type> sel_mat = get_select_matrix<Type>(visits, this->n_visits);
+      matrix<Type> ret = sel_mat * sigma_full * sel_mat.transpose();
+      this->sigmas[visits] = ret;
+      return ret;
+    }
+  }
+  matrix<Type> get_sigma_inverse(std::vector<int> visits, matrix<Type> dist) {
+    auto target = this->simgas_inv.find(visits);
+    if (target != this->simgas_inv.end()) {
```

Comment:
there are typos here - do we have tests in place for this and the other new C++ functions?

Reply:
not yet but we can add it